### PR TITLE
Handle missing/incorrect configs and relation values

### PR DESCRIPTION
**Pull request by jpmartinspt**
Set status to active when everything looks healthy and put the unit in blocked status when there are configuration problems.

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -85,44 +81,49 @@ def _on_controller_relation_changed(self, event):
...
+    def _configure(self):
         """
         Take the configuration values and render them to the index.html file
         for display on the static page.
         """
         data = self._stored.controllerData
+
+        controller_url = data.get("controller-url")
+        identity_provider_url = data.get("identity-provider-url")
+        is_juju = data.get("is-juju", False)
```
Comment (hatched) [inline src/charm.py:98]: I think `True` is a more accurate default here.

File: src/charm.py [hunk 2]
```diff
@@ -85,44 +81,49 @@ def _on_controller_relation_changed(self, event):
...
-            controller_ws_api=data.get("controller-url").replace("wss", "https"),
+            controller_ws_api=controller_url.replace("wss", "https"),
             dashboard_root=os.getcwd()
         ).dump("/etc/nginx/sites-available/default")
 
-    def restart_nginx(self):
-        os.system("sudo service restart nginx")
-
+        nginx_status = os.system("sudo systemctl restart nginx")
+        if nginx_status == 0:
```
Comment (hatched) [inline src/charm.py:123]: Can you add some comments here as to why this check is in here?

File: src/charm.py [hunk 3]
```diff
@@ -85,44 +81,49 @@ def _on_controller_relation_changed(self, event):
...
         env = Environment(loader=FileSystemLoader(os.getcwd()))
         env.filters['bool'] = bool
 
         config_template = env.get_template("src/config.js.template")
         config_template.stream(
-            controller_url=data.get("controller-url"),
             base_app_url="/",
-            identity_provider_url=data.get("identity-provider-url"),
-            is_juju=data.get("is-juju")
+            controller_api_endpoint="/api",
```
Comment (hatched) [inline src/charm.py:110]: Are we using `/api` vs `/controller/api` now?

**Review by hatched — APPROVED** [review]
Looks good, just a few small requests.

File: src/charm.py (hunk 3 above)
Comment (jpmartinspt) [reply src/charm.py:110]: I thought it's just simpler api to revert to `controller/api`

**Pull request merged**